raiden/storage/sqlite.py

```diff
@@ -292,6 +292,42 @@ def get_latest_state_change_by_data_field(
 
         return result
 
+    def get_all_state_changes(self) -> List[StateChangeRecord]:
```

Comment:
IMO it's not a good idea to load a whole database table. This is a bad pattern for large DBs and will result in memory problems (at best, swapping at worst).

For this case we can use sqlite's JSON filtering and preferably work on batches of objects (e.g. an arbitrary batch of 500 state changes)

Reply:
I agree and to be frank I felt dirty by writing that function. I think this https://github.com/raiden-network/raiden/pull/3550/commits/a677e55eb09d1eaeb8202dcf6459dea51a8fabf4 commit improves things. The filtered state change query works fine, but unfortunately for the DB migration I am only using the batch size parameter. That is because I am not aware of a way to do regexp string comparison for the `_type` attribute of the `data` field of state changes.

And a quick look at json sqlite filtering [docs](https://www.sqlite.org/json1.html) did not give a good hint.